mm2src/coins/utxo/utxo_builder/utxo_arc_builder.rs

```diff
@@ -240,42 +242,51 @@ async fn block_header_utxo_loop<T: UtxoCommonOps>(
             Err(e) => {
                 error!("Error {} on getting the height of the last stored header in DB!", e);
                 sync_status_loop_handle.notify_on_temp_error(e.to_string());
-                Timer::sleep(10.).await;
+                Timer::sleep(BLOCK_HEADERS_LOOP_SLEEP_TIMER).await;
                 continue;
             },
         };
 
+        let to_block_height = from_block_height + chunk_size;
+        let to_block_height = if to_block_height > block_count {
+            block_count
+        } else {
+            to_block_height
+        };
+
         // Todo: Add code for the case if a chain reorganization happens
-        if from_block_height == last_block_height {
-            sync_status_loop_handle.notify_sync_finished(last_block_height);
-            last_block_height = match coin.as_ref().rpc_client.get_block_count().compat().await {
-                Ok(h) => h,
-                Err(e) => {
-                    error!("Error {} on getting the height of the latest block from rpc!", e);
-                    sync_status_loop_handle.notify_on_temp_error(e.to_string());
-                    Timer::sleep(10.0).await;
-                    continue;
-                },
```

Comment:
After removing this, when a new header appears on chain we will not be able to fetch it since `from_block_height == block_count` will always be true and the rest of the loop will be bypassed.

Reply:
this pr fixes this issue as I tested it before..new headers will be fetched as supposed if there are new headers(increase in total block count)